TLM/TLM/Util/Extensions/NetSegmentExtensions.cs

```diff
@@ -44,15 +44,24 @@ public static class NetSegmentExtensions {
         }
 
         /// <summary>
-        /// Checks if the netSegment is Created, but neither Collapsed nor Deleted.
+        /// Checks <paramref name="netSegment"/> is <c>Created</c> but not <c>Collapsed|Deleted</c>.
         /// </summary>
         /// <param name="netSegment">netSegment</param>
-        /// <returns>True if the netSegment is valid, otherwise false.</returns>
+        /// <returns>Returns <c>true</c> if valid, otherwise <c>false</c>.</returns>
         public static bool IsValid(this ref NetSegment netSegment) =>
             netSegment.m_flags.CheckFlags(
                 required: NetSegment.Flags.Created,
                 forbidden: NetSegment.Flags.Collapsed | NetSegment.Flags.Deleted);
 
+        /// <summary>
+        /// Checks <paramref name="segmentId"/> is not <c>0</c>,
+        /// then checks netSegment is <c>Created</c> but not <c>Collapsed|Deleted</c>.
+        /// </summary>
+        /// <param name="segmentId">The id of the segment to check.</param>
+        /// <returns>Returns <c>true</c> if valid, otherwise <c>false</c>.</returns>
+        public static bool IsValid(this ushort segmentId)
+            => (segmentId != 0) && segmentId.ToSegment().IsValid();
```

Comment:
checking segmentId != 0 is redundant because m_segments[0] holds the default value and therefore is not valid.

Reply:
I think we should just use the old extension.
If checking `segmentId != 0` is important - for example if you think someone can accidentally write something to `m_segments[0]` - then we should do it consistently everywhere.
But That is OCD in my opinion. the old interface would do.

Reply:
If we're sure lane / segment zero will never have `Created` flag set, then yeah, we could ditch the `id != 0` check.

@krzychu124 thoughts on this?

Reply:
First element is always thrown away on load so it should have flag `None` unless other mod used it and then got serialized to the savegame (bug) (worth checking if they also skip first element or not)

Reply:
So it should be ok to ditch the `id != 0` checks?

Reply:
I would check where CO is checking for `!= 0` -> they know where it is possible. We can only guess

Reply:
> First element is always thrown away on load so it should have flag None unless other mod used it and then got serialized to the savegame (bug) (worth checking if they also skip first element or not)

If that is the case then we should always check for id != 0.
it would be inconsistent to have 2 extensions methods to check if valid and then we pick one or the other randomly.
In any case I consider this to be a low priority issue. I think its OK to merge the code like this but it might be worth improving it.

Reply:
I've realised there are some other issues with this PR, notably that an `IsValid()` extension of a `uint` / `ushort` has no way to determine what class those ids refer to, which is a huge flaw in my thinking of this PR.

That being the case, we should always be doing `.ToWhatever()` first and then `IsValid()`, but is there any way to, for example, get the lane id from a `NetLane` instance?

If we need to check `!= 0` how do we implement that in the least crufty/cumbersome manner?

Reply:
That was what I was afraid of (I should have communicated that better). I thin we should minimize the use of `(interger).Extensions()`. I think `ToNode()` or `ToSegment()` is fine but more than can cause trouble.

Reply:
One thing I was thinking to include `id != 0` check in `IsValid()` extensions is extensions such as `uint.IsValidLane()` and `ushort.IsValidSegment()`, example:

```csharp
internal static bool IsValidLane(this uint laneId)
    => laneId != 0 && laneId.ToLane().IsValid();

// call site example:

if (!someLaneId.IsValidLane())
    return;
```